src/kakarot/execution_context.cairo

```diff
@@ -457,6 +480,38 @@ namespace ExecutionContext {
             evm_contract_address=evm_contract_address,
             calling_context=self.calling_context,
             sub_context=self.sub_context,
+            destroy_contracts_len=self.destroy_contracts_len,
+            destroy_contracts=self.destroy_contracts,
+            );
+    }
+
+    // @notice Update the array of contracts to destroy.
+    // @param self The pointer to the execution context.
+    // @param destroy_contracts_len Array new length.
+    // @param destroy_contracts The pointer to the new array of contracts.
```

Comment:
Because no item in this array is supposed to be removed we may prefer having an update method that only pushes items.